dbt/adapters/spark/impl.py

```diff
@@ -108,53 +115,71 @@ def drop_relation(self, relation, model_name=None):
             kwargs={'relation': relation}
         )
 
-    def get_catalog(self, manifest):
-        schemas = manifest.get_used_schemas()
+    @staticmethod
+    def find_table_information_separator(rows: List[dict]) -> int:
+        pos = 0
+        for row in rows:
+            if not row['col_name']:
```

Comment:
To avoid including partition columns twice, could this be something like
```suggestion
            if not row['col_name'] or row['col_name'] == '# Partition Information':
```
I'm not sure if that's the most elegant way of writing, and it would need at least a line break to satisfy pep8